source/loader/layers/validation/ur_leak_check.hpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include "ur_validation_layer.hpp"
+
+#include <execinfo.h>
+#include <mutex>
+#include <shared_mutex>
+#include <unordered_map>
+#include <utility>
+
+#define MAX_BACKTRACE_FRAMES 64
+
+namespace validation_layer {
+class RefCountContext {
+  private:
+    struct RefRuntimeInfo {
+        int64_t refCount;
+        char **originBacktrace;
+        int backtraceLines;
+    };
+
+    std::shared_mutex mutex;
```

Comment:
you can use std::mutex since you're always using unique_lock with it

Reply:
Done.